botx/client/stickers_api/add_sticker.py

```diff
@@ -0,0 +1,112 @@
+from datetime import datetime as dt
+from typing import NoReturn, Optional
+from uuid import UUID
+
+import httpx
+
+from botx.async_buffer import AsyncBufferReadable
+from botx.client.authorized_botx_method import AuthorizedBotXMethod
+from botx.client.exceptions.http import InvalidBotXStatusCodeError
+from botx.client.stickers_api.exceptions import (
+    InvalidEmojiError,
+    InvalidImageError,
+    StickerPackNotFoundError,
+)
+from botx.models.api_base import UnverifiedPayloadBaseModel, VerifiedPayloadBaseModel
+from botx.models.attachments import (
+    DEFAULT_MIMETYPE,
+    EXTENSIONS_TO_MIMETYPES,
+    encode_rfc2397,
+)
+from botx.models.stickers import Sticker
+
+try:
+    from typing import Literal
+except ImportError:
+    from typing_extensions import Literal  # type: ignore  # noqa: WPS440
+
+
+class BotXAPIAddStickerRequestPayload(UnverifiedPayloadBaseModel):
+    sticker_pack_id: UUID
+    emoji: str
+    image: str
+
+    @classmethod
+    async def from_domain(
+        cls,
+        sticker_pack_id: UUID,
+        emoji: str,
+        async_buffer: AsyncBufferReadable,
+        filename: str,
+    ) -> "BotXAPIAddStickerRequestPayload":
+        file_extension = filename.split(".")[-1]
+        mimetype = EXTENSIONS_TO_MIMETYPES.get(file_extension, DEFAULT_MIMETYPE)
+
+        content = await async_buffer.read()
+        b64_content = encode_rfc2397(content, mimetype)
+
+        return cls(sticker_pack_id=sticker_pack_id, emoji=emoji, image=b64_content)
+
+
+class BotXAPIAddStickerResult(VerifiedPayloadBaseModel):
+    id: UUID
+    emoji: str
+    link: str
+    inserted_at: dt
+    updated_at: dt
+    deleted_at: Optional[dt]
+
+
+class BotXAPIAddStickerResponsePayload(VerifiedPayloadBaseModel):
+    status: Literal["ok"]
+    result: BotXAPIAddStickerResult
+
+    def to_domain(self) -> Sticker:
+        return Sticker(
+            id=self.result.id,
+            emoji=self.result.emoji,
+            image_link=self.result.link,
```

Comment:
Can we use this link to download sticker?

Reply:
Yes, we can